### PR TITLE
added overridden methods equals and hashCode

### PR DESCRIPTION
**Pull request by YarikJ**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -27,4 +29,23 @@ public Integer getLength() {
...
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return this.width * 31 + this.length + this.color.length();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: better don't use values like this: `this.width` and `this.length` and length of the color

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +29,25 @@ public Integer getLength() {
...
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(width, length, color);
+    }
 }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java:53]: remove two blank line

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +29,25 @@ public Integer getLength() {
...
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return Objects.equals(width, rectangle.width)
+                && Objects.equals(length, rectangle.length)
+                && Objects.equals(color, rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(width, length, color);
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: Let’s not use java.util.Objects

**Comment by YarikJ** [comment]
Should I replace  calculation of hashCode with the next statement 
**return length * width * color.hashCode();** ???

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+        Rectangle rectangle = (Rectangle) o;
+        return this.length.equals(rectangle.length)
+                && this.width.equals(rectangle.width)
+                && (this.color == rectangle.color) || (this.color != null
+                && this.color.equals(rectangle.color));
+    }
+
+    @Override
+    public int hashCode() {
+        int colNum = 0;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: can we use a hashcode of width, length and color to calculation hashCode?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,33 @@ public Integer getLength() {
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return this.length.equals(rectangle.length)
+                && this.width.equals(rectangle.width)
+                && (this.color == rectangle.color) || (this.color != null
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: why do we check only color for `null` but not checking other fields?

File: src/main/java/core/basesyntax/Rectangle.java [hunk 6]
```diff
@@ -27,4 +27,28 @@ public Integer getLength() {
...
+                && this.length.equals(rectangle.length)))
+                && ((this.width == rectangle.width) || (this.width != null
+                && this.width.equals(rectangle.width)))
+                && ((this.color == rectangle.color) || (this.color != null
+                && this.color.equals(rectangle.color)));
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * length.hashCode() * width.hashCode() * color.hashCode();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Rectangle.java]: What will happen if one of field, for example length or width == 0?

**Review by Sofasmile — COMMENTED** [review]
Good work! Make small fixing

File: src/main/java/core/basesyntax/Rectangle.java (hunk 6 above)
Comment (YarikJ) [reply src/main/java/core/basesyntax/Rectangle.java]: more collisions would happen

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**